src/mono/wasm/runtime/interpgo.ts

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { Module } from "./globals";
+import { getCacheKey, cleanupCache, getCacheEntry, storeCacheEntry } from "./snapshot";
+import { mono_log_info, mono_log_error } from "./logging";
+import { localHeapViewU8 } from "./memory";
+import cwraps from "./cwraps";
+
+export const tablePrefix = "https://dotnet.generated.invalid/interpgo";
+
+export async function getInterpgoTable(): Promise<ArrayBuffer | undefined> {
+    const cacheKey = await getCacheKey(tablePrefix);
+    if (!cacheKey)
+        return undefined;
+    return await getCacheEntry(cacheKey);
+}
+
+async function storeInterpgoTable(memory: ArrayBuffer) {
+    const cacheKey = await getCacheKey(tablePrefix);
+    if (!cacheKey)
+        return;
+
+    await storeCacheEntry(cacheKey, memory, "application/octet-stream");
+
+    cleanupCache(tablePrefix, cacheKey); // no await
+}
+
+export async function interpgo_save_data () {
+    const cacheKey = await getCacheKey(tablePrefix);
```

Comment:
since this is scheduled 15 seconds later, you can use `loaderHelpers.is_runtime_running` or `loaderHelpers.is_exited` and skip storing it if it's not running. In particular, storing snapshot when something failed (during startup) is probably not good.